Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs

```diff
@@ -349,6 +331,49 @@ private static bool IsNonRetriableException(Exception exception)
                 return false;
             }
 
+            /// <summary>
+            /// Returns an enumerator containing the private and regional service endpoints to iterate over.
+            /// </summary>
+            /// <param name="defaultEndpoint">An instance of <see cref="Uri"/> containing the default global endpoint.</param>
+            /// <param name="locations">An instance of <see cref="IList{T}"/> containing the preferred location names.</param>
+            /// <param name="accountInitializationCustomEndpoints">An instance of <see cref="IList{T}"/> containing the custom private endpoints.</param>
+            /// <returns>An instance of <see cref="IEnumerator{T}"/> containing the service endpoints.</returns>
+            private static IEnumerator<Uri> GetServiceEndpointEnumerator(
+                Uri defaultEndpoint,
+                IList<string>? locations,
+                IList<string> accountInitializationCustomEndpoints)
+            {
+                // We first iterate through all the private endpoints and add them to the service endpoints list.
+                IList<Uri> serviceEndpoints = new List<Uri>();
+
+                if (accountInitializationCustomEndpoints != null
+                    && accountInitializationCustomEndpoints.Count > 0)
+                {
+                    foreach (string customEndpoint in accountInitializationCustomEndpoints)
+                    {
+                        // Add all of the custom private endpoints to the service endpoints list.
+                        serviceEndpoints.Add(
+                            new Uri(customEndpoint));
+                    }
+                }
+
+                // The next step is to construct and add all of the regional endpoints to the service endpoints list.
+                // The regional endpoints will be created by appending the preferred region name as a suffix to the
+                // default global endpoint.
+                if (locations != null

```

Comment:
Since it's nullable, should we use `HasValue` instead?
```suggestion
                if (locations.HasValue && locations.Value.Count > 0

```

Reply:
Looks like, even the nullable collections `<IList<string>? list` doesn't have a definition of `HasValue`. 

![image](https://github.com/Azure/azure-cosmos-dotnet-v3/assets/87335885/3694616e-3504-40b7-9af2-879fa54060a3)